dace/transformation/passes/fusion_inline.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2019-2022 ETH Zurich and the DaCe authors. All rights reserved.
+"""
+Contains implementations of SDFG inlining and state fusion passes.
+"""
+
+from dataclasses import dataclass
+from typing import Any, Dict, Optional
+
+from dace import SDFG
+from dace.sdfg.utils import fuse_states, inline_sdfgs
+from dace.transformation import pass_pipeline as ppl
+
+
+@dataclass(unsafe_hash=True)
+class FuseStates(ppl.Pass):
+    """
+    Fuses all possible states of an SDFG (and all sub-SDFGs).
+    """
+    permissive: bool = False  #: If True, ignores some race conditions checks
+    progress: Optional[bool] = None  #: Whether to print progress, or None for default (print after 5 seconds)
+
+    def should_reapply(self, modified: ppl.Modifies) -> bool:
+        return modified & (ppl.Modifies.States | ppl.Modifies.InterstateEdges)
```

Comment:
Seems like we don't always have a separate test for each pass? As far as I understand, the whole correctness of the compiler now depends on the developer specifying those flags correctly.

Maybe we should stress test each pass in a shotgun approach: Add a test case where we perform randomly generated pipelines on some non-standard SDFGs? Especially the order would be interesting to change